static/app/utils/queryClient.tsx

```diff
@@ -62,7 +62,11 @@ function useQuery<TQueryFnData, TError = RequestError, TData = TQueryFnData>(
     | UseQueryOptions<TQueryFnData, TError, TData>,
   queryOptions?: UseQueryOptions<TQueryFnData, TError, TData>
 ): reactQuery.UseQueryResult<TData, TError> {
-  const api = useApi();
+  // XXX: We need to set persistInFlight to disable query cancellation on unmount.
+  // The current implementation of our API client does not reject on query
+  // cancellation, which causes React Query to never update from the isLoading state.
+  // This matches the library default as well: https://tanstack.com/query/v4/docs/guides/query-cancellation#default-behavior
+  const api = useApi({persistInFlight: true});
```

Comment:
Could we maybe instead add an option to reject on query cancelation? This change might result in a lot of 'state updateed on unmount' type of errors

Reply:
I actually don't think it will do that, React Query says this in the docs:

> By default, queries that unmount or become unused before their promises are resolved are not cancelled. This means that after the promise has resolved, the resulting data will be available in the cache. This is helpful if you've started receiving a query, but then unmount the component before it finishes. If you mount the component again and the query has not been garbage collected yet, data will be available.

I was planning on adding the option to reject as you suggest, but after reading that I convinced myself that it might be better to let the request complete

Reply:
Ah okay cool. Any way we could add a test for this?